### ISSUE TITLE
Virtualbox+Centos: same IP address assigned to two NICs

### ISSUE THREAD

**Issue by EnricoItl** [hashicorp/vagrant#9423]
### Vagrant version
Vagrant 2.0.1

### Host operating system
Windows 10 1709
Virtualbox 5.2.4

### Guest operating system
Centos 6.7

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "bento/centos-6.7"
  config.vm.network :forwarded_port, guest: 22, host: 22111, id: 'ssh'
  config.vm.network "private_network", ip: "192.168.50.51", virtualbox__intnet: true
  config.vm.network "private_network", ip: "192.168.50.52", virtualbox__intnet: true
end
```
### Debug output
https://gist.github.com/EnricoItl/60549fb3990d29abc0ab316b7089986b#file-gistfile1-txt

### Expected behavior
The guest system should start up with eth1 bound to ip 192.168.50.51 and eth2 to 192.168.50.52

### Actual behavior
See debug output.
eth2 has no address assigned because the following script:
mv -f '/tmp/vagrant-network-entry-eth2-1517437391-1' '/etc/sysconfig/network-scripts/ifcfg-eth2'
is wrong:

[vagrant@localhost ~]$ cat /etc/sysconfig/network-scripts/ifcfg-eth2
#VAGRANT-BEGIN
#The contents below are automatically generated by Vagrant. Do not modify.
NM_CONTROLLED=no
BOOTPROTO=none
ONBOOT=yes
IPADDR=192.168.50.51
NETMASK=255.255.255.0
DEVICE=eth2
PEERDNS=no
#VAGRANT-END

It reassigns address 192.168.50.51 which is actually already bound to eth1  

My previous configuration worked fine:
Windows10  1607
Virtualbox 5.1.6
Vagrant 1.8.6
After upgrading Windows to 1709, I had also to upgrade Virtualbox to 5,2,4 and Vagrant to 2.0.1

### Steps to reproduce
Run vagrant up

**Comment by jhkrischel** [comment]
Getting this same issue with Oracle Enterprise Linux 7.4 (another redhat flavor).

**Comment by paulmcgoldrick** [comment]
same. 
### Vagrant version
Vagrant 2.0.2.

### Host operating system
OS X 10.13.2
Version 5.1.22 r115126 (Qt5.6.2)

### Guest operating system
CentOS 7

### Vagrantfile
```ruby
      conf.vm.network :private_network,
        :adapter => 2,
        :ip  => '10.0.0.1',
        :mac => "0800276695A3"

      conf.vm.network :private_network,
        :adapter => 3,
        :ip  => '10.0.0.2',
        :mac => "0800276695A0"
```

### Debug output
Provide a link to a GitHub Gist containing the complete debug output:
https://www.vagrantup.com/docs/other/debugging.html. The debug output should
be very long. Do NOT paste the debug output in the issue, just paste the
link to the Gist.

### Expected behavior
The IP Addresses assigned to these network adapters should correspond to the code above

### Actual behavior
The 2nd network adapter is assigned the first adapters IP address

```sh
vagrant@localhost ~]$ cat /etc/sysconfig/network-scripts/ifcfg-eth1
#VAGRANT-BEGIN
# The contents below are automatically generated by Vagrant. Do not modify.
NM_CONTROLLED=yes
BOOTPROTO=none
ONBOOT=yes
IPADDR=10.0.0.1
NETMASK=255.255.255.0
DEVICE=eth1
PEERDNS=no
#VAGRANT-END


[vagrant@localhost ~]$ cat /etc/sysconfig/network-scripts/ifcfg-eth2
#VAGRANT-BEGIN
# The contents below are automatically generated by Vagrant. Do not modify.
NM_CONTROLLED=yes
BOOTPROTO=none
ONBOOT=yes
IPADDR=10.0.0.1
NETMASK=255.255.255.0
DEVICE=eth2
PEERDNS=no
#VAGRANT-END


[vagrant@localhost ~]$ ip addr show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 52:54:00:ad:3b:43 brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global dynamic eth0
       valid_lft 86342sec preferred_lft 86342sec
    inet6 fe80::5054:ff:fead:3b43/64 scope link
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 08:00:27:66:95:c1 brd ff:ff:ff:ff:ff:ff
    inet 10.0.0.1/24 brd 10.0.0.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::a00:27ff:fe66:95c1/64 scope link
       valid_lft forever preferred_lft forever
4: eth2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 08:00:27:66:95:c2 brd ff:ff:ff:ff:ff:ff
    inet 10.0.0.1/24 brd 10.0.0.255 scope global eth2
       valid_lft forever preferred_lft forever
    inet6 fe80::a00:27ff:fe66:95c2/64 scope link
       valid_lft forever preferred_lft forever
[vagrant@localhost ~]$
```
### Steps to reproduce
Vagrant up

**Comment by EnricoItl** [comment]
Just rechecked with Vagrant 2.1.2: the issue is still there.
Any plan to fix it?
Thanks.

**Comment by chrisroberts** [comment]
Fixed via #10527
